### DISCUSSION TITLE
Error when piping events

### DISCUSSION THREAD

**Discussion by soutomas** [nlmixr2/rxode2#593] [Q&A]
Error when I pipe the event:
![image](https://github.com/cran/rxode2/assets/32617414/7901d56d-19ef-4017-a6e4-0708c64589a4)

However, no issues if I use event in the argument:
![image](https://github.com/cran/rxode2/assets/32617414/2ba5327b-3c29-4381-9347-480656fa7f42)

Not sure why?

**Comment by billdenney** [comment]
I've converted this to a discussion since it's more of a question than an issue.

This is related to piping itself and not related to `rxode2`.  If you rewrote your first example without piping, it would translate to the following:

```
rxSolve(et(m1, amt = 150, until = 48, ii = 24, cmt = 1))
```

Note that the first argument to `rxSolve()` is now `et(...)`.

In your second example, the piping translates to:

```
rxSolve(m1, events = ev)
```

where the model is the first argument to rxSolve.

I'd suggest using the second method for piping your model in.
